File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.js
// Last updated: 4/29/2026, 10:31:44 AM
/**
 * @param {number[]} nums
 * @return {number}
 */
// 2022-10-27
// 2:22s
// O(n)
// O(1)
var findDuplicate = function(nums) {
    const map = {};
    for(const num of nums) {
        if(map[num] === undefined) {
            map[num] = 1;
        } else {
            return num;
        }
    }
};

File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.py
# Last updated: 4/29/2026, 10:31:39 AM
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        from collections import defaultdict
        dict = defaultdict(lambda: None)
        for num in nums:
            if dict[num] == True:
                return num
            else:
                dict[num] = True